### PR TITLE
Add notes about safe uses of twox

### PR DESCRIPTION
**Pull request by DemiMarie-parity**
to help with future security auditing.

### CODE REVIEW

File: frame/elections/src/lib.rs [hunk 1]
```diff
@@ -237,16 +237,25 @@ decl_storage! {
...
 		pub RegisterInfoOf get(fn candidate_reg_info):
 			map hasher(twox_64_concat) T::AccountId => Option<(VoteIndex, u32)>;
 		/// Basic information about a voter.
+		///
+		/// TWOX-NOTE: SAFE as `AccountId` is a crypto hash.
 		pub VoterInfoOf get(fn voter_info):
 			map hasher(twox_64_concat) T::AccountId => Option<VoterInfo<BalanceOf<T>>>;
 		/// The present voter list (chunked and capped at [`VOTER_SET_SIZE`]).
+		///
+		/// TWOX-NOTE: OKAY ― `SetIndex` is not controlled by an attacker.
```
Comment (NikVolf) [inline frame/elections/src/lib.rs:258]: would avoid "attacker" here

potential adversary? or just "data is not user-controlled"

File: frame/babe/src/lib.rs [hunk 2]
```diff
@@ -149,6 +149,8 @@ decl_storage! {
 		/// We reset all segments and return to `0` at the beginning of every
 		/// epoch.
 		SegmentIndex build(|_| 0): u32;
+
+		/// NOTE-TWOX: `SegmentIndex` is an increasing integer, so this is okay.
```
Comment (NikVolf) [inline frame/babe/src/lib.rs:153]: TWOX-NOTE or NOTE-TWOX ?

File: frame/grandpa/src/lib.rs [hunk 3]
```diff
@@ -212,6 +212,8 @@ decl_storage! {
 
 		/// A mapping from grandpa set ID to the index of the *most recent* session for which its
 		/// members were responsible.
+		///
+		/// TWOX-NOTE: `SetId` is not under the controll of an attacker.
```
Comment (NikVolf) [inline frame/grandpa/src/lib.rs:216]:
```suggestion
		/// TWOX-NOTE: `SetId` is not under user control.
```

File: frame/elections/src/lib.rs (hunk 1 above)
Comment (gavofyork) [reply frame/elections/src/lib.rs:258]:
```suggestion
		/// TWOX-NOTE: OKAY ― `SetIndex` is not user-controlled data.
```

File: frame/assets/src/lib.rs [hunk 4]
```diff
@@ -257,6 +257,8 @@ decl_storage! {
 		/// The next asset identifier up for grabs.
 		NextAssetId get(fn next_asset_id): T::AssetId;
 		/// The total unit supply of an asset.
+		///
+		/// NOTE-TWOX: `AssetId` is trusted, so this is safe.
```
Comment (gavofyork) [inline frame/assets/src/lib.rs:261]:
```suggestion
		/// TWOX-NOTE: `AssetId` is trusted, so this is safe.
```

File: frame/babe/src/lib.rs (hunk 2 above)
Comment (gavofyork) [reply frame/babe/src/lib.rs:153]:
```suggestion
		/// TWOX-NOTE: `SegmentIndex` is an increasing integer, so this is okay.
```

**Pull request merged**